feat(eks): support Kubernetes 1.21

jumic opened the pull request:
Support Kubernetes 1.21

Closes #15758.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: db6b33760c927c5d19cddbb0ad249d281e517dd6
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c9e08de1-409b-4271-9857-a729ea265978%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

otaviomacedo commented on packages/@aws-cdk/lambda-layer-kubectl/layer/Dockerfile:
```diff
@@ -5,7 +5,7 @@ FROM public.ecr.aws/lambda/provided:latest
 #
 
-ARG KUBECTL_VERSION=1.20.0
+ARG KUBECTL_VERSION=1.21.0
```
Version 1.21 of `kubectl` is not compatible with versions of the server lower than 1.21. We haven't come up with a long term strategy for it yet. For now, please leave this version as it is.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 4bc7bcc094916664e58cec0de36d3e7dc6b7bbe2
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=60d900f2-2b11-4161-8860-90e6a531954e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 988e6f3d6fe11ccc5e053ed8aecd471a5b584e8c
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=761856ea-5271-4ed9-aa2d-9c2b5813cc03%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

jumic commented:
@otaviomacedo Thanks for your feedback. I reverted the version upgrade in the Dockerfile of lambda-layer-kubectl. In addition I added a comment in the Dockerfile with reference to issue #15736 to avoid the same change in future until there is a solution for supporting different versions in parallel.
I hope this is fine for you.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: f0bca63d5289aa64e5f232374c5e425798499b27
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e89b1559-a806-4451-bce4-9ceb5f5091e7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->
